HIP 107: Clarifications and Recommendations (#923)
helium/HIP: 0107-preventing-gaming-within-the-mobile-network.md

Outdoor Radios suspected of gaming may have their CPI registration revoked or denied, and will not be able to earn rewards until a new submission is submitted and approved.

### Outdoor Wi-Fi Maximum Height
It is worth noting that due to the shorter range of outdoor Wi-Fi, the maximum asserted height will be changed to 100 feet. This will be enforced initially in MOBILE oracles by ignoring values over 100 feet and may, in the future be blocked at the protocol level. Deployments can still be deployed higher than 100 feet but these values will not be considered for the purpose of modeled coverage rewards.

### Maximum Asserted Distance Difference

> For indoor and outdoor Wi-Fi access points, Hotspot locations are verified using external services like Skyhook. If the asserted distance is greater than 300 meters away from the external service location, the trust score will be set to `0.00`.

Author: ```suggestion
For indoor and outdoor Wi-Fi access points, Hotspot locations are verified using external services like Skyhook. If the asserted distance is greater than 1000 meters away from the external service location, the trust score will be set to `0.00`.
```

Author: Recommended update directly from @heatedlime

> 

Reviewer: Can you add:

Further, in order for access points to be eligible for boosted Service Provider rewards, the asserted distances must be 30 meters or less than the external service location. In instances where the difference is greater than 30 meters, the access point will still be eligible for PoC rewards, but not Service Provider boosted rewards.

Author: sure. I can.

> 

Author: ```suggestion

Can you add:

Further, in order for access points to be eligible for boosted Service Provider rewards defined in HIP-84, the asserted distances must be 30 meters or less than the reported location from external services. In instances where the difference is greater than 30 meters, the access point will still be eligible for PoC rewards, but not boosted rewards.
```

    - Consistently offloading data of subscribers who signed up with fraudulent information, such as payment source

## Abuse of Power
If it is identified that Service Providers are abusing their power to turn off MOBILE equipment without valid gaming reasons, or do not adhere to the requirements of this HIP, a governance proposal may be raised by the MOBILE Working Group to slash an amount of the Service Providers stake. The proposal must be voted on by the MOBILE network with veMOBILE and pass by at least 67%.

## Drawbacks

Deployers who have an outdoor Wi-Fi Access Point asserted higher than 100 feet will need to update the assertion height to a maximum of 100 feet.

## Rationale and Alternatives
An alternative would be to do nothing and allow gaming to continue.

## Implementation Phases

For HIP implementation, Helium Mobile and Nova Labs has agreed to do the following:

- Update the Helium Builder app to require a picture be uploaded when new assertions are done for outdoor Wi-Fi access points. This feature also needs to be backported to the Helium Wallet app since assertions can occur in this application.

> - Update the Hotspot trust score for Wi-Fi Hotspots to `0.00` if the asserted location is 300 meters or more from external data sources like Skyhook, etc.

Author: ```suggestion
- Update the Hotspot trust score for Wi-Fi Hotspots to `0.00` if the asserted location is 1000 meters or more from external data sources like Skyhook, etc.
```

Outdoor Radios suspected of gaming may have their CPI registration revoked or denied, and will not be able to earn rewards until a new submission is submitted and approved.

### Outdoor Wi-Fi Maximum Height
It is worth noting that due to the shorter range of outdoor Wi-Fi, the maximum asserted height will be changed to 100 feet. This will be enforced initially in MOBILE oracles by ignoring values over 100 feet and may, in the future be blocked at the protocol level. Deployments can still be deployed higher than 100 feet but these values will not be considered for the purpose of modeled coverage rewards.

### Maximum Asserted Distance Difference
For indoor and outdoor Wi-Fi access points, Hotspot locations are verified using external services like Skyhook. If the asserted distance is greater than 1000 meters away from the external service location, the trust score will be set to `0.00`.

> Further, in order for access points to be eligible for boosted Service Provider rewards defined in HIP-84, the asserted distances must be 30 meters or less than the reported location from external services. In instances where the difference is greater than 30 meters, the access point will still be eligible for PoC rewards, but not boosted rewards.

Author: ```suggestion
Further, in order for access points to be eligible for boosted Service Provider rewards defined in HIP-84, the asserted distances must be 30 meters or less than the reported location from external services. In instances where the difference is greater than 30 meters, the access point will still be eligible for PoC rewards, but not boosted rewards.

```

Specifically, Service Providers may stop radios/access points from producing heartbeats if they are suspected of gaming, effectively setting their reward multiplier to `0X` for both PoC and Data Transfer rewards. This may be set on CBRS Radios and Wi-Fi access points for up to ten (`10`) consecutive epochs. The specific reasons are identified in the section below. The reason and length of deactivation must be publicly submitted to the MOBILE Oracle. After the defined deactivation time (10 epochs maximum), the device must be automatically returned to normal participation on the network (with the exception of outdoor CBRS radios, which need a newly approved CPI submission). However, if the gaming activity continues, the Service Provider may deactivate the device again using the same duration defined above.

Outdoor Radios suspected of gaming may have their CPI registration revoked or denied, and will not be able to earn rewards until a new submission is submitted and approved.

### Outdoor Wi-Fi Maximum Height

> It is worth noting that due to the shorter range of outdoor Wi-Fi, the maximum asserted height will be changed to 100 feet. This will be enforced initially in MOBILE oracles by ignoring values over 100 feet and may, in the future be blocked at the protocol level. Deployments can still be deployed higher than 100 feet but these values will not be considered for the purpose of modeled coverage rewards.

Reviewer 2: "ignore" also used in this sentence

> It is worth noting that due to the shorter range of outdoor Wi-Fi, the maximum asserted height will be changed to 100 feet. This will be enforced initially in MOBILE oracles by ignoring values over 100 feet and may, in the future be blocked at the protocol level. Deployments can still be deployed higher than 100 feet but these values will not be considered for the purpose of modeled coverage rewards.

Reviewer 3: @madninja how is height currently submitted for wifi hotspots? There's no concept of elevation on-chain for mobile, only IOT.

Author: We should scope adding it to outdoor WiFi Hotspots.

    - Height mismatch
    - Location mismatch
    - Spoofing of location

- False, inaccurate, or deceptive images/data submitted

- Fraud
    - Including subscriber abuse of subscribers signing up, offloading data, and then canceling the subscriber plan prior to grace period end date
    - Consistently offloading data of subscribers who signed up with fraudulent information, such as payment source

## Abuse of Power
If it is identified that Service Providers are abusing their power to turn off MOBILE equipment without valid gaming reasons, or do not adhere to the requirements of this HIP, a governance proposal may be raised by the MOBILE Working Group to slash an amount of the Service Providers stake. The proposal must be voted on by the MOBILE network with veMOBILE and pass by at least 67%.

## Drawbacks

Deployers who have an outdoor Wi-Fi Access Point asserted higher than 100 feet will need to update the assertion height to a maximum of 100 feet.

## Rationale and Alternatives
An alternative would be to do nothing and allow gaming to continue.

## Implementation Phases

For HIP implementation, Helium Mobile and Nova Labs has agreed to do the following:

> - Update the Helium Builder app to require a picture be uploaded when new assertions are done for outdoor Wi-Fi access points. This feature also needs to be backported to the Helium Wallet app since assertions can occur in this application.

Reviewer 3: The helium wallet app needs to be updated to require this as well. And the elevation.

Author: I think it says that but open to edits.

- Update the Hotspot trust score for Wi-Fi Hotspots to `0.00` if the asserted location is 1000 meters or more from external data sources like Skyhook, etc.
- Update the Helium Builder app to change the maximum height an outdoor Wi-Fi Access Point can be asserted to at 100 feet.

> - Update the MOBILE oracle to ignore any heights asserted by outdoor Wi-Fi access points higher than 100 feet.

Reviewer 2: be more precise about "ignore".. does that mean the previous height is to be used or (what I think you're saying) is to use 100 feet if the assertion is hight than 100 feet. Two places in this doc that refer to to "ignoring"

Author: updated to be more clear.

> - Update the MOBILE oracle to ignore any heights asserted by outdoor Wi-Fi access points higher than 100 feet.

Author: ```suggestion
- Update the MOBILE oracle to use 100 feet for any heights asserted by outdoor Wi-Fi access points higher than this value.
```

> - Enhance the CPI process to submit deactivation reason and deadline to MOBILE oracles

Reviewer 2: what does this mean? The oracles don't _use_ the CPI deactivation reason and deadline at all today, where would this go and who would use it?

Reviewer 2: Sorry, misread, I get it.. this is.. a bit more work

Author: @heatedlime can probably provide more details here on why this is needed. It was in the original HIP that I just did some reformatting.

Reviewer 2: this part is definitely a decent chunk of work to add to the system.. not impossible, but perhaps do this HIP in phases?

Author: Reasonable. @heatedlime how would you phase this out?

Reviewer: Hows this @abhay @madninja 

Phase 1
-  Create a process for Service Providers to submit deactivation reason and deadline to MOBILE Oracles _(Since CPI rejections are communicated via email and none of the CPI process is on chain, I don't think it's worth incorporating it into here)_
- Update MOBILE oracles to automatically reactivate devices after the specified deadline.

Phase 2
- Update the MOBILE oracle to use 100 feet for any heights asserted by outdoor Wi-Fi access points higher than this value.
- Enforce maximum elevation for Outdoor Wi-Fi Hotspots on chain and make this configurable.
- Update the Helium Builder app to change the maximum height an outdoor Wi-Fi Access Point can be asserted to at 100 feet.

Phase 3
- Update the Hotspot trust score for Wi-Fi Hotspots to 0.00 if the asserted location is 1000 meters or more from external data sources like Skyhook, etc.
- Update the Helium Builder app to require a picture be uploaded when new assertions are done for outdoor Wi-Fi access points. This feature also needs to be backported to the Helium Wallet app since assertions can occur in this application.
- Update the protocol to add elevation to on chain data of MOBILE hotspots.

Reviewer: @abhay based on discussion with @madninja @ChewingGlass in the 107 channel, here’s the updated list for phases 

Phase 1

- Create a process for Service Providers to submit deactivation reason and deadline to MOBILE Oracles 
- Update MOBILE oracles to automatically reactivate devices after the specified deadline.

Phase 2

- Update the MOBILE oracle to use 100 feet for any heights asserted by outdoor Wi-Fi access points higher than this value.
- Enforce maximum elevation for Outdoor Wi-Fi Hotspots on chain and make this configurable.
- Update the Helium Builder app to change the maximum height an outdoor Wi-Fi Access Point can be asserted to at 100 feet.

Phase 3

- Update the Hotspot trust score for Wi-Fi Hotspots to 0.00 if the asserted location is 1000 meters or more from external data sources like Skyhook, etc.
- Update the Helium Builder app to require a picture be uploaded when new assertions are done for outdoor Wi-Fi access points.
- Update and backfill the protocol to add Height and Azimuth to on chain data of MOBILE hotspots.
- Update the Helium Wallet app to allow location assertions within the app by capturing height, azimuth, and deployment photo

Author: I can add that to this PR for you @heatedlime

Reviewer: > I can add that to this PR for you @heatedlime

Yes please

> - Update MOBILE oracles to automatically reactivate devices after the specified deadline.

Reviewer 3: There's likely additional protocol work required to add `elevation` to mobile hotspots. As well as enforce a maximum.

> - Update the protocol to add `elevation` to on chain data of MOBILE hotspots.

Reviewer: Azimuth should also be added.

> - Update MOBILE oracles to automatically reactivate devices after the specified deadline.

Author: ```suggestion
- Update MOBILE oracles to automatically reactivate devices after the specified deadline.
- Update the protocol to add `elevation` to on chain data of MOBILE hotspots.
- Enforce maximum elevation for Outdoor Wi-Fi Hotspots on chain and make this configurable. 
```

- Radio/Wi-Fi AP is not in the same location as the submission data states. This includes:
    - Height mismatch
    - Location mismatch
    - Spoofing of location

- False, inaccurate, or deceptive images/data submitted

- Fraud
    - Including subscriber abuse of subscribers signing up, offloading data, and then canceling the subscriber plan prior to grace period end date
    - Consistently offloading data of subscribers who signed up with fraudulent information, such as payment source

## Abuse of Power
If it is identified that Service Providers are abusing their power to turn off MOBILE equipment without valid gaming reasons, or do not adhere to the requirements of this HIP, a governance proposal may be raised by the MOBILE Working Group to slash an amount of the Service Providers stake. The proposal must be voted on by the MOBILE network with veMOBILE and pass by at least 67%.

## Drawbacks

Deployers who have an outdoor Wi-Fi Access Point asserted higher than 100 feet will need to update the assertion height to a maximum of 100 feet.

## Rationale and Alternatives
An alternative would be to do nothing and allow gaming to continue.

## Implementation Phases

> 
> For HIP implementation, Helium Mobile and Nova Labs has agreed to do the following:

Author: ```suggestion

For HIP implementation, Helium Mobile and Nova Labs has agreed to do the following:
```
